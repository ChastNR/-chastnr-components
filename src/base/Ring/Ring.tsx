import classNames from 'clsx';
import { ReactComponent as RingSvg } from './circle.svg';

import './Ring.scss';

interface RingProps {
  color?: string;
  size?: 'small' | 'large';
}

const Ring: React.FC<RingProps> = ({ color = '#1771e6', size = 'small' }) => {
  const ringClassNames = classNames('ring', {
    'ring--small': size === 'small',
  });

  return (
    <div className={ringClassNames}>
      <RingSvg stroke={color} />
    </div>
  );
};

export { Ring, RingSvg };
