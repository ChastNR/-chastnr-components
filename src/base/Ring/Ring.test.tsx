import { render } from '@testing-library/react';
import { Ring } from './Ring';

describe('<Calendar/>', () => {
  it('should equal snapshot', () => {
    const ring = render(<Ring />);
    expect(ring.container.firstChild).toMatchSnapshot();
  });
});
