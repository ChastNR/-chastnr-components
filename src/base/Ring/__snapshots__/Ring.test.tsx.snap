// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar/> should equal snapshot 1`] = `
<div
  class="ring ring--small"
>
  <span>
    content
  </span>
</div>
`;
